### PR TITLE
working towards fuller docstring coverage (atm 50%)

### PR DESCRIPTION
**Pull request by razofz**
Mostly written docstrings, some grammatical changes in docs pages as well. 

Output of `docstr-coverage -mP ./scarf`:
`Overall statistics for 19 files (skipped all non-init magic methods) (skipped private methods):
Needed: 280  -  Found: 142  -  Missing: 138
Total coverage: 50.7%  -  Grade: Not bad`

Not sure what to write for the meld_assay module, don't understand exactly what it does yet.

### CODE REVIEW

**Comment by razofz** [comment]
ping @parashardhapola

File: scarf/__init__.py [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+- Perform quick and accurate projections of cells from one dataset onto another or integrate multiple datasets.
+
+Exports:
+--------
+
+- Modules
+    - datastore: Contains the primary interface to interact with data (i. e. DataStore) and its superclasses.
+    - readers: A collection of classes for reading in different data formats.
+    - writers: Methods and classes for writing data to disk.
+    - meld_assay:
```
Comment (razofz) [inline scarf/__init__.py:22]: not sure what to write for meld_assay

**Review by parashardhapola — APPROVED** [review]

File: scarf/__init__.py (hunk 1 above)
Comment (parashardhapola) [reply scarf/__init__.py:22]: I can take are of that!

**Comment by parashardhapola** [comment]
This is great. Let's see how it renders on readthedocs' API section.

**Pull request merged**